ivy/functional/backends/jax/experimental/statistical.py

```diff
@@ -301,11 +302,7 @@ def cummax(
     dtype: Optional[jnp.dtype] = None,
     out: Optional[JaxArray] = None,
 ) -> Tuple[JaxArray, JaxArray]:
-    if x.dtype in (jnp.bool_, jnp.float16):
-        x = x.astype(jnp.float64)
-    elif x.dtype in (jnp.int16, jnp.int8, jnp.uint8):
-        x = x.astype(jnp.int64)
-    elif x.dtype in (jnp.complex128, jnp.complex64):
+    if x.dtype in (jnp.complex128, jnp.complex64):
         x = jnp.real(x).astype(jnp.float64)
```

Comment:
I think here we should just allow this casting for `complex128` and mark `complex64` as unsupported because if we cast the real element of `complex64` (which is `float32`) to `float64`, we're having the same kind of additional memory consumption as with casting all `int` values to `int64` and so on. 
I think the same has happened in the PRs we merged yesterday. I couldn't notice that then but seems like doing this would be preferable, what do you think?

Reply:
sounds good, I'll make sure that every `complex64` casting is removed.

Reply:
Ved you were right I think we can get the real part without creating a new array
```python
a = np.random.uniform(size=10)
a.dtype
Out[49]: dtype('float64')
a.astype("complex128")
Out[50]: 
array([0.53129972+0.j, 0.16200054+0.j, 0.08087117+0.j, 0.76545754+0.j,
       0.51599048+0.j, 0.37456005+0.j, 0.16486012+0.j, 0.77161369+0.j,
       0.92052105+0.j, 0.90555509+0.j])
ac = a.astype("complex128")
ac.real
Out[53]: 
array([0.53129972, 0.16200054, 0.08087117, 0.76545754, 0.51599048,
       0.37456005, 0.16486012, 0.77161369, 0.92052105, 0.90555509])
ac.real[0] = 123
ac
Out[55]: 
array([**_1.23000000e+02+0.j_**, 1.62000537e-01+0.j, 8.08711702e-02+0.j,
       7.65457537e-01+0.j, 5.15990475e-01+0.j, 3.74560048e-01+0.j,
       1.64860120e-01+0.j, 7.71613695e-01+0.j, 9.20521046e-01+0.j,
       9.05555095e-01+0.j])
```